compiler-rt/lib/sanitizer_common/sanitizer_procmaps_mac.cpp

```diff
@@ -334,9 +334,22 @@ static const load_command *NextCommand(const load_command *lc) {
   return (const load_command *)((const char *)lc + lc->cmdsize);
 }
 
-static void FindUUID(const load_command *first_lc, u8 *uuid_output) {
-  for (const load_command *lc = first_lc; lc->cmd != 0; lc = NextCommand(lc)) {
-    if (lc->cmd != LC_UUID) continue;
+#ifdef MH_MAGIC_64
+static const size_t header_size = sizeof(mach_header_64);
+#else
+static const size_t header_size = sizeof(mach_header);
+#endif
+
+static void FindUUID(const load_command *first_lc, const mach_header *hdr,
+                     u8 *uuid_output) {
+  unsigned short curcmd = 0;
```

Comment:
Any specific reason this a short?  I usually use the "natural" / register-width-sized `size_t` for any sizes or counters.
Or at least `uint32_t`, because `hdr->ncmds` is `uint32_t`.

Reply:
Done. Went with uint32_t so it has same type as HDR->ncmds